src/util/hasher.h

```diff
@@ -8,6 +8,7 @@
 #include <crypto/siphash.h>
 #include <primitives/transaction.h>
 #include <uint256.h>
+#include <chain.h>
```

Comment:
it *might* be better to forward declare CBlockIndex and then move the BlockHasher constructor to the .cpp file, making `mock` a unique_ptr instead... that way we don't need to pull chain in anywhere it's needed. But then there's overhead!

Another option would be to template BlockHasher with T and then fill in CBlockIndex where it's used locally.

Reply:
Yes , Now i have defined BlockHasher as a Template and filled it with CBlockIndex in validation.h when required.